src/fhir/api.ts

```diff
@@ -33,45 +34,68 @@ export const bundleRequest = (form: FormValues, reportedGenes: RequiredCoding[])
 };
 
 export const createBundle = (form: FormValues, reportedGenes: RequiredCoding[]) => {
-  const org = organisationAndId(form.address);
-  const patient = patientAndId(form.patient, org.id);
-  const specimen = specimenAndId(form.sample, patient.id);
-  const furtherTesting = furtherTestingAndId(form.result, patient.id);
-  const plan = planDefinitionAndId(form.sample, form.result, patient.id);
-  const { authoriser, reporter } = practitionersAndIds(form.result);
-  let variants: ResourceAndIds[];
+  const org = organisationAndIdentifier(form.address);
+  const patient = patientAndIdentifier(form.patient, org.identifier);
+  const specimen = specimenAndIdentifier(form.sample, patient.identifier);
+  const furtherTesting = furtherTestingAndId(form.result, patient.identifier);
+  const plan = planDefinitionAndId(form.sample, form.result, patient.identifier);
+  const { authoriser, reporter } = practitionersAndQueries(form.result);
+  let variants: ResourceAndIdentifier[];
   if (form.variant.length) {
     variants = form.variant.map((variant: VariantSchema) =>
-      variantAndId(variant, reportedGenes, patient.id, specimen.id, specimen.identifier, reporter.id, authoriser.id),
+      variantAndIdentifier(
+        variant,
+        reportedGenes,
+        patient.identifier,
+        specimen.identifier,
+        specimen.identifier,
+        reporter.identifier,
+        authoriser.identifier,
+      ),
```

Comment:
ah yes that because we're using a query based on the identifier rather than the specific id. Turns out the batch bundle on azure means that it expects the id to match what is in the fhir database, whereas the hapi transactional bundle uses the id for the references but accepts it if they are different to what exists in the database